### PR TITLE
Add Waveshare RP2040-GEEK + ESP32-S3-GEEK feature

### PR DESCRIPTION
**Pull request by RetiredWizard**
Add newly merged board https://github.com/adafruit/circuitpython/pull/9090#pullrequestreview-1969157331

Also updated the Waveshare ESP32-S3-GEEK page to include the Display feature.

### CODE REVIEW

**Review by makermelissa — APPROVED** [review]
Thank you

File: _board/waveshare_rp2040_geek.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+name: "RP2040-GEEK"
+manufacturer: "Waveshare"
+board_url:
+ - "https://www.waveshare.com/rp2040-geek.htm"
+board_image: "waveshare_rp2040_geek.jpg"
+date_added: 2024-03-29
+family: raspberrypi
+features:
+  - USB-C
+  - Wi-Fi
```
Comment (jvprat) [inline _board/waveshare_rp2040_geek.md:14]: Is this correct? The RP2040 doesn't have WiFi, is it provided externally?

File: _board/waveshare_rp2040_geek.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+title: "RP2040-GEEK Download"
+name: "RP2040-GEEK"
+manufacturer: "Waveshare"
+board_url:
+ - "https://www.waveshare.com/rp2040-geek.htm"
+board_image: "waveshare_rp2040_geek.jpg"
+date_added: 2024-03-29
+family: raspberrypi
+features:
+  - USB-C
```
Comment (jvprat) [inline _board/waveshare_rp2040_geek.md:13]: AFAIK none of these boards (esp32s3 and rp2040) have USB-C

Comment (RetiredWizard) [reply _board/waveshare_rp2040_geek.md:13]: Ugh, I don't know where my head was, good catch, I'll submit a new PR to drop the RP2040 WiFi and remove USB-C from both the RP2040 and S3 Geek modules.

Thanks!

**Pull request merged**